client/components/Home/index.tsx

```diff
@@ -1,32 +1,85 @@
-import React, { ReactElement } from "react";
-import Typography from "@material-ui/core/Typography";
-
+/* eslint-disable prettier/prettier */
+import React, { ReactElement, useState } from "react";
+import { Link } from "react-router-dom";
+import Axios from "axios";
 import { useHomeStyles } from "./styles";
 
-import Feed from "./Feed";
-import RecentVideos from "./RecentVideos";
-import Film from "./FeaturedMovie";
-
 const Home = (): ReactElement => {
   const classes = useHomeStyles({});
+  const [item, setItem] = useState(null);
+  const [refresh, setRefresh] = useState(true);
+  const [itemReady, setItemReady] = useState(false);
+
+  const getRecommendation = (): void => {
+    const list: Array<object> = [];
+    for (let index = 0; index < 4; index++) {
+      Axios(`https://tv-v2.api-fetch.website/movies/1`).then((res) => {
+        const id = Math.round(Math.random() * 50);
+        console.log(id);
+        list.push({
+          title: res.data[id].title,
+          img: res.data[id].images.poster,
+          imdbId: res.data[id].imdb_id,
+          genres: res.data[id].genres,
+          src: `http://localhost:8080/movie/${res.data[id].imdbId}`
+        });
+        setItem(list);
+        if (index === 3) {
+          setItemReady(true);
+        }
+      });
+    }
+  };
```

Comment:
For the `getRecommendation` function, I have a lot to say:

1. Putting the Axios request **IN** the for loop is a **terrible idea** !! 😱

You're doing 4 time the same request to get the same result ! 😱 Just putting the axios request out of the loop is far more efficient: You do one request Then, you select 4 random movies in the first page
```typescript
  const getRecommendation = (): void => {
    const list: Array<object> = [];
    Axios(`https://tv-v2.api-fetch.website/movies/1`).then((res) => {
      for (let index = 0; index < 4; index++) {
        const id = Math.round(Math.random() * 50);
        list.push({
          title: res.data[id].title,
          img: res.data[id].images.poster,
          imdbId: res.data[id].imdb_id,
          genres: res.data[id].genres
        });
      }
      setItem(list);
    });
  };
```
(I've also put the `setItem(list)` out of the for loops to do it once and not 4th)

--------------
2. Random can select the same movie two times

It happened to me once, it was weird and created an error in the console because two elements had the same `key` property (yeah, I change the code as I comment, my code is using the `map`).
You can check if an id is already like that:
```typescript
item.filter((m) => m.imdbId === id)
```
if it return an empty array, the id is good

----------
3. You should put that in an other file

We try put all logic that is not relative rendering the page in other files (like gating data, checking user input...). In general we name it `[component name].service.tsx`. For you it would be `Home.service.tsx`.

------------
4. In fact, I lied, we should put that in the back 🤭

I don't think we should be doing call to other api's than ours from the front.
You should create a `/movies/recommended` route that return some recommended moives.
The back should then call others api, like the pop corn time one.

Then calling our api from the front is really easy, we created the `useApi` hook to do it !

(It could be used a bit like that, to give you an idea:)
```typescript
const { resData, error } = useApi<Movies[], void>("/movies/recommended", {
  hotReload: true
});
```

--------------
5. On selecting random movies

I find it weird that every time you recharge the home page, the movies are different. Maybe we should take the trending movies or the ones with the most download, or the last one te be added.
Or we could use a seed with the random so that the random movies change every... hour ? but in the same hour, the movies are the same for everyone.